docs/source/tutorial/interfaces.rst

```diff
@@ -264,11 +262,11 @@ abstracts the notion of function application:
 
 .. _monadsdo:
 
-Monads and ``do``-notation
+Monads and ``do`` notation
```

Comment:
This should be reverted throughout. It might not look the prettiest, but writing "`do`-notation" is common in the literature.

Reply:
I think it makes sense to use a hyphen when not using a code font: "do-notation". But as code, ``do`` behaves like a noun, thus technically would not warrant hyphenation. I do get that it might look more familiar with a hyphen, though. Perhaps it would make more sense to simply write "do-notation", without code font?